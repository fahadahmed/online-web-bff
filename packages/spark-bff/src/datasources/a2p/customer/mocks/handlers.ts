import { getAdminOverviewMock } from './getAdminOverview';
import { getCustomerOverviewMock } from './getCustomerOverview';
import { getContentProvidersMock } from './getContentProviders';
import { getServiceRequestsMock } from './getServiceRequests';
import { getShortcodesMock } from './getShortcodes';
import { getShortcodeMock } from './getShortcode';
import { getUsageMock } from './getUsage';
import { addContentProviderMock } from './addContentProvider';
import { addServiceRequestMock } from './addServiceRequest';
import { modifyShortcodeMock } from './modifyShortcode';
import { modifyContentProviderMock } from './modifyContentProvider';
import { deleteContentProviderMock } from './deleteContentProvider';

const handlers = [
  addContentProviderMock,
  addServiceRequestMock,
  getAdminOverviewMock,
  getCustomerOverviewMock,
  getContentProvidersMock,
  getServiceRequestsMock,
  getShortcodesMock,
  getShortcodeMock,
  getUsageMock,
  modifyShortcodeMock,
  modifyContentProviderMock,
  deleteContentProviderMock,
];

export default handlers;
