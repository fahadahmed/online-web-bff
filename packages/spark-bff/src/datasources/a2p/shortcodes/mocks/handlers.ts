import { getAdminShortcodesMock } from './getShortcodes';
import { getAdminShortcodesOverviewMock } from './getShortcodesOverview';

const handlers = [getAdminShortcodesMock, getAdminShortcodesOverviewMock];

export default handlers;
