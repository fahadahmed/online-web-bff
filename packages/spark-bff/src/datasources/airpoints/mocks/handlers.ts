import { rest } from 'msw';
import airpointsResponseMock from './airpoints-response.mock.json';
import addAirpointsResponseMock from './add-airpoints-response.mock.json';
import deleteAirpointsResponseMock from './delete-airpoints-response.mock.json';
import airpointsInactiveResponseMock from './airpoints-inactive-response.mock.json';
import addAirpointsErrorResponseMock from './add-airpoints-error-response.mock.json';
import deleteAirpointsErrorResponseMock from './delete-airpoints-error-response.mock.json';

interface AddAirpointsMockBody {
  accountNumber?: string;
  airpointsNumber?: string;
  firstName?: string;
  lastName?: string;
  status?: string;
}

export const airpointsResponseMocks = rest.get<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber === '123456789') {
      /* Simulate error responses */
      return res(ctx.status(200), ctx.json(airpointsResponseMock));
    }
    return res(ctx.status(200), ctx.json(airpointsInactiveResponseMock));
  },
);

export const updateAirpointsResponseMocks = rest.post<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async (
    { params: { accountNumber }, body: { airpointsNumber } },
    res,
    ctx,
  ) => {
    /* add airpoints */
    if (airpointsNumber) {
      if (airpointsNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(400), ctx.json(addAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(addAirpointsResponseMock));
    }
    /* delete airpoints */
    if (accountNumber) {
      if (accountNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(200), ctx.json(deleteAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(deleteAirpointsResponseMock));
    }
    return null;
  },
);

const handlers = [airpointsResponseMocks, updateAirpointsResponseMocks];

export default handlers;
