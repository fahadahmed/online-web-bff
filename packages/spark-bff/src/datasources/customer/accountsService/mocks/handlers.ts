import { rest } from 'msw';
import accountSummaryMock from './account-summary.mock.json';
import emptyAccountListMock from './empty-account-list.mock.json';
import singleAccountListMock from './single-account-list.mock.json';
import accountListMock from './account-list.mock.json';
import guestAccountListMock from './guest-account-list.mock.json';
import unauthorizedAccountListMock from './unauthorized-account-list.mock.json';

const baseUrl = 'http://testhost/v1/customer/account';

export const accountSummaryHandler = rest.get<unknown, { accountId: string }>(
  `${baseUrl}/:accountId/summary`,
  async (req, res, ctx) => {
    const { accountId } = req.params;
    if (accountId.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountSummaryMock));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

export const accountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(accountListMock));
  },
);

export const singleAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(singleAccountListMock));
  },
);

export const emptyAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(emptyAccountListMock));
  },
);

export const unauthorizedAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(401), ctx.json(unauthorizedAccountListMock));
  },
);

export const guestAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(403), ctx.json(guestAccountListMock));
  },
);

const handlers = [accountSummaryHandler, accountListHandler];

export default handlers;
