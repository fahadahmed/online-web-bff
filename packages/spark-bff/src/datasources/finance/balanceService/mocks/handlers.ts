import { rest } from 'msw';
import accountBalanceDetails from './account-balance.mock.json';

const baseUrl = 'http://testhost/v1/finance/balance/';

export const accountBalanceMock = rest.get<unknown, { accountNumber: string }>(
  `${baseUrl}account/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountBalanceDetails));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

const handlers = [accountBalanceMock];

export default handlers;
