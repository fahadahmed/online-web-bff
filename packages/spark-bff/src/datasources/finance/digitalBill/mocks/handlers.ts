import { rest } from 'msw';
import accountBillingSummary from './account-billing-summary.mock.json';
import accountTransactions from './account-transactions.mock.json';

const baseUrl = 'http://testhost/v2/finance';

export const accountBillingSummaryMock = rest.get(
  `${baseUrl}/:accountNumber/bills/summary`,
  async (_, res, ctx) => {
    return res(ctx.json(accountBillingSummary));
  },
);

export const accountTransactionsMock = rest.get(
  `${baseUrl}/:accountNumber/bills/transactions`,
  async (_, res, ctx) => {
    return res(ctx.json(accountTransactions));
  },
);

const handlers = [accountBillingSummaryMock, accountTransactionsMock];

export default handlers;
