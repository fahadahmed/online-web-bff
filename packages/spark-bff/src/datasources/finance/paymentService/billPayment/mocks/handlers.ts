import { rest } from 'msw';
import poliPayBillSuccess from './poliPayBill-success.mock.json';
import poliPayBillValidationFailure from './poliPayBill-failure.mock.json';
import creditCardPayBillSuccess from './creditCard-payBill-success.mock.json';
import creditCardPayBillValidationFailure from './creditCard-payBill-failure.mock.json';
import deleteAutoPayBillSuccess from './delete-autoPayBill-success.mock.json';
import deleteAutoPayBillFailure from './delete-autoPayBill-failure.mock.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const addPoliPayBillMock = rest.post<unknown, { accountNumber: string }>(
  `${baseUrl}/polipay/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if ((accountNumber as string).match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(poliPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(poliPayBillSuccess));
  },
);

export const addCreditCardPayBillMock = rest.post<
  unknown,
  { accountNumber: string }
>(
  `${baseUrl}/creditcard/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if (accountNumber.match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(creditCardPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(creditCardPayBillSuccess));
  },
);

export const deleteAutoPayBillMock = rest.delete<
  unknown,
  { accountNumber: string; paymentMethodId: string }
>(
  `${baseUrl}/bill/:accountNumber/paymentsetup/:paymentMethodId`,
  async ({ params: { accountNumber, paymentMethodId } }, res, ctx) => {
    /** If accountNumber does not match the given pattern,
     for example, 746738998.
     Or If paymentMethodId does not match the given pattern,
     for example,"1-S0123-23"
     it returns failure response, "Request Validation Failure".
     */
    if (
      accountNumber.match(/^\d+$/) === null ||
      paymentMethodId.match(/^[\dA-Z][\dA-Z-]+[\dA-Z]$/) === null
    ) {
      return res(ctx.status(400), ctx.json(deleteAutoPayBillFailure));
    }
    return res(ctx.status(200), ctx.json(deleteAutoPayBillSuccess));
  },
);

const handlers = [
  addCreditCardPayBillMock,
  addPoliPayBillMock,
  deleteAutoPayBillMock,
];

export default handlers;
