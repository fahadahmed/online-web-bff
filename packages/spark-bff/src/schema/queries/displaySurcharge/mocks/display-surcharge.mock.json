{
  "PAY_BILL": {
    "operationType": "PAY_BILL",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "American Express and Diners Club incur a 2% surcharge.",
      "Visa/MasterCard, Prezzy Card, international credit, debit or charge cards incur a 3% surcharge."
    ]
  },
  "SETUP_MONTHLY_PAYMENT": {
    "operationType": "SETUP_MONTHLY_PAYMENT",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "Automated payments using a credit/debit card incurs a 1.5% surcharge (plus GST, if any). Surcharge applies."
    ]
  }
}
