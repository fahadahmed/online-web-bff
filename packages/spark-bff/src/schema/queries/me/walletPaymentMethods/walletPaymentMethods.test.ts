import { walletPaymentMethodMock } from 'datasources/finance/walletService/mocks/handlers';
import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';

beforeEach(() => {
  server.use(walletPaymentMethodMock);
});

test('returns cardDetails and bankAccountDetails', async () => {
  server.use(walletPaymentMethodMock);
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          paymentMethods {
            type
            source
            paymentMethodId
            personalisedName
            isPreferred
            associations {
              type
              accessType
              accountNumber
              lineNumber
              paymentMethodId
            }
            cardDetail {
              ...card
            }
            bankAccountDetail {
              ...bank
            }
          }
        }
      }

      fragment card on CardDetail {
        maskedCardNumber
        cardName
        cardType
        expiryMonth
        expiryYear
        creditCardType
        brand
      }

      fragment bank on BankAccountDetail {
        bankName
        accountName
        accountNumber
        bankType
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.paymentMethods).toMatchSnapshot();
});

test('returns payment methods filtered by sources', async () => {
  server.use(walletPaymentMethodMock);
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query PaymentMethods($sources: [WalletPaymentMethodSource!]) {
        me {
          paymentMethods(sources: $sources) {
            type
            source
            paymentMethodId
          }
        }
      }
    `,
    variables: {
      sources: ['DIRECT', 'WALLET_SELF'],
    },
  });
  expect(res.errors).toBeUndefined();
  expect(res.data!.me.paymentMethods).toMatchSnapshot();
});

test('returns payment methods filtered by associated line number', async () => {
  server.use(walletPaymentMethodMock);
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query PaymentMethods(
        $sources: [WalletPaymentMethodSource!]
        $associatedLineNumber: String
      ) {
        me {
          paymentMethods(
            sources: $sources
            associatedLineNumber: $associatedLineNumber
          ) {
            type
            source
            paymentMethodId
          }
        }
      }
    `,
    variables: {
      sources: ['DIRECT'],
      associatedLineNumber: '0278675180',
    },
  });
  expect(res.errors).toBeUndefined();
  expect(res.data!.me.paymentMethods).toMatchSnapshot();
});
