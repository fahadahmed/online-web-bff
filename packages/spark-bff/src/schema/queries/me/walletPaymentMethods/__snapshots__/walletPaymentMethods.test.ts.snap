// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns cardDetails and bankAccountDetails 1`] = `
Array [
  Object {
    "associations": Array [
      Object {
        "accessType": "LINE",
        "accountNumber": null,
        "lineNumber": "0278675180",
        "paymentMethodId": "MySpark-SSC1463625391900736-721",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": null,
    "cardDetail": Object {
      "brand": "mastercard",
      "cardName": "Karthick CC",
      "cardType": "MASTER CARD",
      "creditCardType": "mastercard",
      "expiryMonth": 5,
      "expiryYear": 2025,
      "maskedCardNumber": "512345******2346",
    },
    "isPreferred": true,
    "paymentMethodId": "MySpark-SSC1463625391900736-721",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "associations": null,
    "bankAccountDetail": null,
    "cardDetail": Object {
      "brand": "visa",
      "cardName": "Karthick CC",
      "cardType": "VISA CARD",
      "creditCardType": "visa",
      "expiryMonth": 5,
      "expiryYear": 2025,
      "maskedCardNumber": "512345******3425",
    },
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1463625391900736-722",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "associations": null,
    "bankAccountDetail": null,
    "cardDetail": Object {
      "brand": "amex",
      "cardName": "Karthick CC",
      "cardType": "AMEX",
      "creditCardType": "amex",
      "expiryMonth": 5,
      "expiryYear": 2025,
      "maskedCardNumber": "512345******3521",
    },
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1463625391900736-723",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "associations": null,
    "bankAccountDetail": null,
    "cardDetail": Object {
      "brand": "diners",
      "cardName": "Karthick CC",
      "cardType": "DINERS CARD",
      "creditCardType": "diners",
      "expiryMonth": 5,
      "expiryYear": 2025,
      "maskedCardNumber": "512345******3111",
    },
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1463625391900736-725",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196000",
      "bankName": "ANZ Bank New Zealand",
      "bankType": "ANZ_Bank",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1463625391900736-724",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196001",
      "bankName": "Bank of New Zealand",
      "bankType": "Bank_of_New_Zealand",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1433625391900736-724",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196002",
      "bankName": "Westpac",
      "bankType": "Westpac",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1463643391900736-724",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196003",
      "bankName": "Industrial and Commercial Bank of China (New Zealand) Ltd",
      "bankType": "Industrial_and_Commercial_Bank_of_China_Limited_NZ",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC12363625391900736-724",
    "personalisedName": null,
    "source": "WALLET_OTHER",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196004",
      "bankName": "ASB Bank",
      "bankType": "ASB_Bank",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC14636835391900736-724",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196005",
      "bankName": "TSB Bank",
      "bankType": "TSB_Bank",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC146362500900736-724",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196006",
      "bankName": "HSBC New Zealand",
      "bankType": "HSBC_Bank_NZ",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1463625389700736-724",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": "448159183",
        "lineNumber": null,
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196007",
      "bankName": "Citibank N.A.",
      "bankType": "Citibank_N_A",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC14609625391900736-724",
    "personalisedName": null,
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "associations": Array [
      Object {
        "accessType": "ACCOUNT",
        "accountNumber": null,
        "lineNumber": "0278675180",
        "paymentMethodId": "1-ST0123",
        "type": "RECURRING",
      },
    ],
    "bankAccountDetail": Object {
      "accountName": "K D DUNN",
      "accountNumber": "1539590055196008",
      "bankName": "Kiwibank",
      "bankType": "Kiwibank",
    },
    "cardDetail": null,
    "isPreferred": false,
    "paymentMethodId": "MySpark-SSC1463687391900736-724",
    "personalisedName": null,
    "source": "DIRECT",
    "type": "BANK",
  },
]
`;

exports[`returns payment methods filtered by associated line number 1`] = `
Array [
  Object {
    "paymentMethodId": "MySpark-SSC1463687391900736-724",
    "source": "DIRECT",
    "type": "BANK",
  },
]
`;

exports[`returns payment methods filtered by sources 1`] = `
Array [
  Object {
    "paymentMethodId": "MySpark-SSC1463625391900736-721",
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1463625391900736-722",
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1463625391900736-723",
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1463625391900736-725",
    "source": "WALLET_SELF",
    "type": "CREDIT",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1463625391900736-724",
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1433625391900736-724",
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1463643391900736-724",
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "paymentMethodId": "MySpark-SSC14636835391900736-724",
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "paymentMethodId": "MySpark-SSC146362500900736-724",
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1463625389700736-724",
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "paymentMethodId": "MySpark-SSC14609625391900736-724",
    "source": "WALLET_SELF",
    "type": "BANK",
  },
  Object {
    "paymentMethodId": "MySpark-SSC1463687391900736-724",
    "source": "DIRECT",
    "type": "BANK",
  },
]
`;
